tests/test_nodes_utils.py

```diff
@@ -40,3 +48,105 @@ def test_consume_common_prefix(left, right, expected):
     expected_b = (expected[0], expected[2], expected[1])
     assert actual_a == expected
     assert actual_b == expected_b
+
+
+@pytest.mark.parametrize(
+    'valid_node',
+    (
+        (b"\x00\x03\x04\x05\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x02value"),  # noqa: E501
+    ),
+)
+def test_binary_trie_node_parsing_with_valid_input(valid_node):
+    parse_node(valid_node)
+
+
+@pytest.mark.parametrize(
+    'invalid_node',
+    (
+        (b''),
+        (None),
+        (b"\x00\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\x02\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b'\x02'),
+    ),
+)
+def test_binary_trie_node_parsing_with_invalid_input(invalid_node):
+    with pytest.raises(InvalidNode):
+        parse_node(invalid_node)
+
+
+@pytest.mark.parametrize(
+    'keypath,valid_value',
+    (
+        (b'\x00', b"\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+    ),
+)
+def test_encode_binary_trie_kv_node_with_valid_input(keypath, valid_value):
+    encode_kv_node(keypath, valid_value)
```

Comment:
Same suggestion here about adding some assertions about the value returned from this function.